tests/functional/syntax/test_getters.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from vyper.compiler import compile_code
+from vyper.exceptions import SyntaxException
+
+
+def test_duplicate_public_annotation():
+    code = """
+a: public(public(uint256))
```

Comment:
how about `transient(transient(uint256))`?

or for that matter, `transient(constant(immutable(uint256)))` :thinking:

Reply:
that's protected here https://github.com/vyperlang/vyper/pull/4623/files#diff-ffed91cdb4980c8aae7f25d57bb11b5095755e9b951a1b89b2aa71bbb2cb1bb5R1430

Reply:
added test cases in ce8cd27bdc6d80d3b4c2ad235fd09782e931dd23